tests/datasets/test_synthetic_data.py

```diff
@@ -5,7 +5,81 @@
 import pytest
 
 from composer.datasets.synthetic import (SyntheticBatchPairDataset, SyntheticDataLabelType, SyntheticDataType,
-                                         SyntheticPILDataset)
+                                         SyntheticHFDataset, SyntheticPILDataset, generate_synthetic_tokenizer)
+
+
+@pytest.mark.parametrize('num_samples', [50])
+@pytest.mark.parametrize('chars_per_sample', [128])
+@pytest.mark.parametrize('column_names', [['sentence'], ['sentence1', 'sentence2']])
+@pytest.mark.parametrize('tokenizer_family', ['bert', 'gpt2'])
+def test_synthetic_hf_dataset_creation(num_samples: int, chars_per_sample: int, column_names: list,
```

Comment:
Recommend breaking up all the asserts here to individual tests, using fixtures for setup, e.g.
```
@pytest.mark.parametrize('num_samples', [50])
@pytest.mark.parametrize('chars_per_sample', [128])
@pytest.mark.parametrize('column_names', [['sentence'], ['sentence1', 'sentence2']])
@pytest.mark.parametrize('tokenizer_family', ['bert', 'gpt2'])
class TestSyntheticNLP:

   @pytest.fixture
    def dataset_generator(...)
         return SyntheticHFDatset(..)
    
    @pytest.fixture
    def dataset(dataset_generator):
        return dataset_generator.generate_dataset()

    def test_generator_sample(dataset_generator, chars_per_sample):
        sample = dataset_generator.generate_sample()
        assert len(sample) == chars_per_sample

    def test_dataset_shape(dataset, ...):
        assert len(dataset) ==
        assert len(datsaet[..) == 
```

And so forth for the individual tokenizer tests, etc. Essentially everywhere you have a comment below should be converted into an isolated test.



```

Reply:
Yeah, I agree -- thanks for pointing this out, I didn't know about this!

Reply:
Resolving this since it is done